app/models/allocation.rb

```diff
@@ -51,7 +53,11 @@ def initialize(job:, nodes:)
 
   def nodes
     @nodes ||= @node_names.map do |node_name|
-      FlightScheduler.app.nodes.fetch_or_add(node_name)
+      FlightScheduler.app.nodes[node_name].tap do |node|
+        raise MissingNodeError, <<~ERROR.chomp if node.nil?
+          Tried to allocate missing node: '#{node_name}'
+        ERROR
+      end
```

Comment:
In general, creating an allocation with a missing node is an error condition. However an optional `add` flag has been added to handle the above edge case due to the `persistence`: https://github.com/openflighthpc/flight-scheduler-controller/commit/b8e01726e5f35698db1a01ee0a8947a6c538c4db

I can't think of any other situations this will occur, as such they can be dealt with if and when a problem arises.

Reply:
You've used `map` where I think you should have used `tap`.  It's still broken.

Reply:
Good spot, turns out it is neither:
https://github.com/openflighthpc/flight-scheduler-controller/commit/e5d61701254720251a1138aaba71e0abb2fe32fc